### PR TITLE
Added view for creating new member and showing details of member

### PR DESCRIPTION
**Pull request by papperwing**

### CODE REVIEW

File: persistence/src/main/java/cz/muni/fi/pa165/entity/Member.java [hunk 1]
```diff
@@ -29,7 +29,6 @@
     @Column(nullable = false)
     private String givenName;
 
-    @Column(nullable = false)
```
Comment (msimacek) [inline persistence/src/main/java/cz/muni/fi/pa165/entity/Member.java:32]: Why?

File: api/src/main/java/cz/muni/fi/pa165/dto/MemberRegisterDTO.java [hunk 2]
```diff
@@ -7,9 +7,9 @@
  */
 public class MemberRegisterDTO {
 
-    private MemberDTO member;
+    private MemberDTO member = new MemberDTO();
```
Comment (msimacek) [inline api/src/main/java/cz/muni/fi/pa165/dto/MemberRegisterDTO.java:10]: It should contain just id, it doesn't need a DTO

File: web/src/main/webapp/META-INF/context.xml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Context antiJARLocking="true" path=""/>
```
Comment (msimacek) [inline web/src/main/webapp/META-INF/context.xml:2]: Why do you think this file is necessary?

File: sample-data/src/main/java/cz/muni/fi/pa165/sample/SampleDataLoader.java [hunk 4]
```diff
@@ -53,11 +57,18 @@ private void createMember() {
...
-        memberRegister.setPassword("12345");
-        memberFacade.registerMember(memberRegister);
+        memberRegister.setPassword("12345");       
+        Long memberId = memberFacade.registerMember(memberRegister);
+        createLoan(memberId);
+    }
+    
+    private void createLoan(Long memberId){
+        CreateLoanDTO loan = new CreateLoanDTO();
+        loan.setBookId(1L);
```
Comment (msimacek) [inline sample-data/src/main/java/cz/muni/fi/pa165/sample/SampleDataLoader.java:70]: It might not be 1L, don't rely on DB implementation details. Pass the id as argument.

File: persistence/src/main/java/cz/muni/fi/pa165/entity/Member.java (hunk 1 above)
Comment (papperwing) [inline persistence/src/main/java/cz/muni/fi/pa165/entity/Member.java:32]: Because I guess, that we don't want profiles without password

File: api/src/main/java/cz/muni/fi/pa165/dto/MemberDTO.java [hunk 5]
```diff
@@ -54,9 +54,9 @@ public void setEmail(String email) {
     public Date getRegistrationDate() {
         return registrationDate;
     }
-
-    public void setRegistrationDate(Date registrationDate) {
-        this.registrationDate = registrationDate;
+    
+    public void setRegistrationDate(Date date) {
+        this.registrationDate = date;
```
Comment (msimacek) [inline api/src/main/java/cz/muni/fi/pa165/dto/MemberDTO.java:59]: Please don't make unnecessary changes like this, it just increases the number of possible conflicts with others

File: persistence/src/main/java/cz/muni/fi/pa165/entity/Member.java (hunk 1 above)
Comment (msimacek) [inline persistence/src/main/java/cz/muni/fi/pa165/entity/Member.java:32]: Indeed, but this is not the password field

File: web/src/main/webapp/WEB-INF/jsp/member/create.jsp [hunk 6]
```diff
@@ -0,0 +1,22 @@
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8"%>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core"%>
+<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form"%>
+<%@ taglib tagdir="/WEB-INF/tags" prefix="x"%>
+<x:base title="Register new Member">
+    <jsp:attribute name="content">
+        <h1>Register new member</h1>
+        <form:form method="POST" modelAttribute="createMember">
+            <div class="form-group">
+                <form:input path="member.givenName" type="text" placeholder="*Username" cssClass="form-control" />
```
Comment (msimacek) [inline web/src/main/webapp/WEB-INF/jsp/member/create.jsp:10]: Use labels, not placeholders. Labels don't disappear after you started typing and work in all browsers

File: web/src/main/webapp/WEB-INF/jsp/member/create.jsp [hunk 7]
```diff
@@ -0,0 +1,22 @@
+<%@ page language="java" contentType="text/html; charset=UTF-8" pageEncoding="UTF-8"%>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core"%>
+<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form"%>
+<%@ taglib tagdir="/WEB-INF/tags" prefix="x"%>
+<x:base title="Register new Member">
```
Comment (msimacek) [inline web/src/main/webapp/WEB-INF/jsp/member/create.jsp:5]: I'd use different title. This page is supposed to be used by a user to register himself, not someone else. So something like "Sing up" or "Register"

File: web/src/main/webapp/WEB-INF/jsp/member/create.jsp (hunk 6 above)
Comment (msimacek) [inline web/src/main/webapp/WEB-INF/jsp/member/create.jsp:10]: And it's not username, it's given name

File: web/src/main/webapp/WEB-INF/jsp/member/create.jsp [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+                <form:input path="member.givenName" type="text" placeholder="*Username" cssClass="form-control" />
+                <form:errors path="member.givenName" />
+                <form:input path="member.surname" type="text" placeholder=" Surname" cssClass="form-control" />
+                <form:errors path="member.surname" />
+                <form:input path="member.email" type="email" placeholder="*E-mail" cssClass="form-control" />
+                <form:errors path="member.email" />
+                <form:input path="password" type="password" placeholder="*Password" cssClass="form-control" />
+                <form:errors path="password" />
+            </div>
+            <button type="submit" class="btn btn-default">Register!</button>
```
Comment (msimacek) [inline web/src/main/webapp/WEB-INF/jsp/member/create.jsp:19]: No "!", please

File: web/src/main/webapp/WEB-INF/jsp/member/show.jsp [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<x:base>
+    <jsp:attribute name="title">
+        <c:out value="${member.givenName}" /> - member detail
+    </jsp:attribute>
+    <jsp:attribute name="content">
+        <h1>
+            <c:out value="${member.givenName}" />
+        </h1>
+        ID:<c:out value="${member.id}" />
```
Comment (msimacek) [inline web/src/main/webapp/WEB-INF/jsp/member/show.jsp:13]: Some styling please, no <br/>.

File: web/src/main/webapp/WEB-INF/jsp/member/show.jsp [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core"%>
+<%@ taglib tagdir="/WEB-INF/tags" prefix="x"%>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<x:base>
+    <jsp:attribute name="title">
+        <c:out value="${member.givenName}" /> - member detail
+    </jsp:attribute>
+    <jsp:attribute name="content">
+        <h1>
+            <c:out value="${member.givenName}" />
```
Comment (msimacek) [inline web/src/main/webapp/WEB-INF/jsp/member/show.jsp:11]: Surname should also go here

File: web/src/main/webapp/WEB-INF/jsp/member/show.jsp [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+                        <th>Returned</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    <c:forEach items="${loans}" var="loan">
+                        <tr>
+                            <td><c:out value="${loan.id}" /></td>
+                            <td><c:out value="${loan.book.name}"/></td>
+                            <td><fmt:formatDate value="${loan.loanDate}" pattern="yyyy-MM-dd"/></td>
+                            <td>${loan.returned ? 'Returned' : 'Loaned'}</td>
```
Comment (msimacek) [inline web/src/main/webapp/WEB-INF/jsp/member/show.jsp:35]: It would be better to display two separate tables: one for already returned loans and one for currently active loans (with expected return dates).
